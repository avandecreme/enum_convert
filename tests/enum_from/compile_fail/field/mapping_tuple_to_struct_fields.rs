use enum_convert::EnumFrom;

enum Source {
    Tuple(i32, i32),
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from(Source::Tuple)]
    Struct {
        #[enum_from(Source::Tuple.0)]
        a: i32,
        #[enum_from(Source::Tuple.1)]
        b: i32,
    },
}

fn main() {}
