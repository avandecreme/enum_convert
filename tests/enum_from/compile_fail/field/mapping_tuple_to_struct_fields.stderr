error: Unexpected mapping to positional field for struct variant
  --> tests/enum_from/compile_fail/field/mapping_tuple_to_struct_fields.rs:12:35
   |
12 |         #[enum_from(Source::Tuple.0)]
   |                                   ^
